src/csharp/Grpc.Tools/build/_protobuf/Google.Protobuf.Tools.targets

```diff
@@ -28,6 +28,7 @@
       <ProtoRoot Condition="'%(Protobuf.ProtoRoot)' == '' " />
       <CompileOutputs Condition="'%(Protobuf.CompileOutputs)' == ''">True</CompileOutputs>
       <OutputDir Condition="'%(Protobuf.OutputDir)' == '' ">$(Protobuf_OutputPath)</OutputDir>
+      <Generator Condition=" '$(DisableProtobufDesignTimeBuild)' != 'true' " >MSBuild:Compile</Generator>
```

Comment:
Does it make sense to follow the pattern and use 
```xml
<Generator Condition=" %(Protobuf.Generator)' == '' and '$(DisableProtobufDesignTimeBuild)' != 'true' " >MSBuild:Compile</Generator>
```
? If the generator can change per item, why not allow setting it using a user's itemdefgroup?

Reply:
I missed that.

Reply:
Sure, it's kind of half-way in the middle. If the generator may never change at all, or should be global, then a property would work better. But as long as it is among the item's metadata, the usual pattern is to allow a redefinition of defaults for the whole ItemGroup.

I still had no time to play with MSBuild to see of a simpler syntax of list augmentation would cut it. Likely tomorrow.